src/legacy/ui/public/agg_types/controls/date_ranges.tsx

```diff
@@ -0,0 +1,184 @@
+  to?: string;
+}
+
+interface DateRangeValuesModel extends DateRangeValues {
+  id: string;
+}
+
+function DateRangesParamEditor({
+  value = [],
+  setValue,
+  setValidity,
+}: AggParamEditorProps<DateRangeValues[]>) {
+  const [ranges, setRanges] = useState(() => value.map(range => ({ ...range, id: generateId() })));
+
+  // set up an initial range when there is no default range
+  useEffect(() => {
+    if (!value.length) {
+      onAddRange();
+    }
+  }, []);
+
+  useEffect(
+    () => {
+      // responsible for discarding changes
+      if (
+        value.length !== ranges.length ||
+        value.some((range, index) => !isEqual(range, omit(ranges[index], 'id')))
+      ) {
+        setRanges(value.map(range => ({ ...range, id: generateId() })));
+      }
+
+      setValidity(value.every(({ from, to }) => validateDateMath(from) && validateDateMath(to)));
+    },
+    [value]
+  );
+
+  const updateRanges = (rangeValues: DateRangeValuesModel[]) => {
+    // do not set internal id parameter into saved object
+    setValue(rangeValues.map(range => omit(range, 'id')));
+    setRanges(rangeValues);
+  };
+  const onAddRange = () => updateRanges([...ranges, { id: generateId() }]);
+  const onRemoveRange = (id: string) => updateRanges(ranges.filter(range => range.id !== id));
+  const onChangeRange = (id: string, key: string, newValue: string) =>
+    updateRanges(
+      ranges.map(range =>
+        range.id === id
+          ? {
+              ...range,
+              [key]: newValue === '' ? undefined : newValue,
+            }
+          : range
+      )
+    );
+
+  return (
+    <>
+      {ranges.map(({ from, to, id }) => (
+        <Fragment key={id}>
+          <EuiFlexGroup gutterSize="s" alignItems="center">
+            <EuiFlexItem>
+              <EuiFieldText
+                aria-label={i18n.translate('common.ui.aggTypes.dateRanges.fromColumnLabel', {
+                  defaultMessage: 'From',
+                  description:
+                    'Describes the beginning of a date range, e.g. *From* 2018-02-26 To 2018-02-28',
+                })}
+                value={isEmpty(from) ? '' : from}
+                onChange={ev => onChangeRange(id, 'from', ev.target.value)}
+                fullWidth={true}
+                isInvalid={!validateDateMath(from)}
+              />
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiIcon type="sortRight" color="subdued" />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiFieldText
+                aria-label={i18n.translate('common.ui.aggTypes.dateRanges.toColumnLabel', {
+                  defaultMessage: 'To',
+                  description:
+                    'Describes the end of a date range, e.g. From 2018-02-26 *To* 2018-02-28',
+                })}
+                value={isEmpty(to) ? '' : to}
+                onChange={ev => onChangeRange(id, 'to', ev.target.value)}
+                fullWidth={true}
```

Comment:
Can you make these input compressed?

Reply:
`EuiFieldText` can't be compressed until #38262 will be merged